### PR TITLE
[infra] helper.py: support "profile" command for a single fuzz target with arbitrary corpus location.

### PR DESCRIPTION
**Pull request by Dor1s**
I've implemented this in a way such that it should be easy to extend to any number of fuzz targets specified, if there will be a demand for that usecase.

### CODE REVIEW

File: infra/helper.py [hunk 1]
```diff
@@ -541,31 +545,39 @@ def download_corpus(project_name):
...
+    print('ERROR: --corpus-dir argument cannot be used without --fuzz-target',
+          file=sys.stderr)
+    return 1
   if not _check_project_exists(args.project_name):
     return 1
-  if not args.no_corpus_download:
-    if not download_corpus(args.project_name):
+  if not args.no_corpus_download and not args.corpus_dir:
+    if not download_corpus(args):
      return 1
```
Comment (inferno-chromium) [inline infra/helper.py]: is the indent correct, seems off-by-one.

File: infra/helper.py [hunk 2]
```diff
@@ -541,31 +545,39 @@ def download_corpus(project_name):
...
   return not False in thread_pool.map(_download_for_single_target, fuzz_targets)
 
 
 def profile(args):
   """Generate code coverage using clang source based code coverage."""
+  if args.corpus_dir and not args.fuzz_target:
+    print('ERROR: --corpus-dir argument cannot be used without --fuzz-target',
+          file=sys.stderr)
+    return 1
   if not _check_project_exists(args.project_name):
```
Comment (inferno-chromium) [inline infra/helper.py]: new line before this line

File: infra/helper.py [hunk 3]
```diff
@@ -541,31 +545,39 @@ def download_corpus(project_name):
...
     return 1
-  if not args.no_corpus_download:
-    if not download_corpus(args.project_name):
+  if not args.no_corpus_download and not args.corpus_dir:
```
Comment (inferno-chromium) [inline infra/helper.py]: new line before this line

File: infra/helper.py [hunk 4]
```diff
@@ -541,31 +545,39 @@ def download_corpus(project_name):
...
-  print('Downloading corpus for %s project' % project_name)
+  print('Downloading corpus for %s project' % args.project_name)
   thread_pool = ThreadPool(multiprocessing.cpu_count())
   return not False in thread_pool.map(_download_for_single_target, fuzz_targets)
 
 
 def profile(args):
   """Generate code coverage using clang source based code coverage."""
+  if args.corpus_dir and not args.fuzz_target:
+    print('ERROR: --corpus-dir argument cannot be used without --fuzz-target',
```
Comment (inferno-chromium) [inline infra/helper.py]: Lets rephrase this, i got confused reading it (cannot and without, not not :). --corpus-dir requires specifying a particular fuzz target using --fuzz-target

File: infra/helper.py (hunk 1 above)
Comment (Dor1s) [reply infra/helper.py]: Indeed! Done.

File: infra/helper.py (hunk 2 above)
Comment (Dor1s) [reply infra/helper.py]: Done.

File: infra/helper.py (hunk 3 above)
Comment (Dor1s) [reply infra/helper.py]: Done.

File: infra/helper.py (hunk 4 above)
Comment (Dor1s) [reply infra/helper.py]: :D thanks for the suggestion! I should also update the doc page I've added recently.

**Pull request merged**